docs/baseband/gsb/index.rst

```diff
@@ -9,6 +9,224 @@ the `Giant Metrewave Radio Telescope (GMRT) <http://www.gmrt.ncra.tifr.res.in/>`
+has a single data file and accompanying header representing the voltage stream
+from one polarization of a single dish.  The header is just the computer-based
+timestamp (accurate to :math:`1\,\mathrm{\mu s}`)::
+
+    YYYY MM DD HH MM SS.SSSSSS
+
+Rawdump is recorded at a rate of 33.3333... megasamples per second (Msps). 
+Each sample is 4 bits in size, and two samples are grouped into 8-bit words
+such that the oldest sample occupies the least significant bit.  Each data
+frame consists of **4 megabytes** worth, or :math:`2^{23}`, samples; as such,
+the timespan of one frame is exactly **251.658240 ms**.
+
+Data in phased format is spread over multiple binary files and one
+accompanying header file.  The right and left circular polarization data are
+separated into two separate sets of files.  Each polarization is further
+subdivided into multiple files with staggered timestreams such that each file
+contains a portion of every frame.  For example, a typical GSB observation may
+contain four files, two for each polarization.  For each polarization, the
+first half of the payload data of each frame is stored in one file, and the
+second half is stored in the other.
+
+The "raw", or pre-channelized, :term:`sample rate` is either 33.3333... Msps at
+8 bits per sample or 66.6666... Msps at 4 bits per sample (in the latter case,
+sample bit-ordering is the same as for rawdump).  Channelization via fast
+Fourier transform sets the channelized :term:`complete sample` rate to the raw
+rate divided by :math:`2N_\mathrm{F}`, where :math:`N_\mathrm{F}` is the
+number of Fourier channels. The timespan of one frame is **251.658240 ms**, and
+one frame is **8 megabytes** in size, for either raw sample rate.
+
+The phased header's structure is::
+
+    <PC TIME> <GPS TIME> <SEQ NUMBER> <MEM BLOCK>
+
+where ``<PC TIME>`` and ``<GPS TIME>`` are the less accurate computer-based
+and exact GPS-based timestamps, respectively, with the same format as the
+rawdump timestamp; ``<SEQ NUMBER>`` is the frame number; and ``<MEM BLOCK>``
+a redundant modulo-8 shared memory block number not used in analysis.
+
+.. _gsb_usage:
+
+Usage Notes
+===========
+
+This section covers reading and writing GSB files with Baseband; general usage
+is covered in the :ref:`Using Baseband <using_baseband_toc>` section.
+The examples below use the sample files in the ``baseband/data/gsb/``
+directory, and the `numpy`, `astropy.units` and `baseband.gsb` modules::
+
+    >>> import numpy as np
+    >>> import astropy.units as u
+    >>> from baseband import gsb
+    >>> from baseband.data import (
+    ...     SAMPLE_GSB_RAWDUMP, SAMPLE_GSB_RAWDUMP_HEADER,
+    ...     SAMPLE_GSB_PHASED, SAMPLE_GSB_PHASED_HEADER)
+
+A single timestamp file can be opened with `~baseband.gsb.open` in text
+mode::
+
+    >>> ft = gsb.open(SAMPLE_GSB_RAWDUMP_HEADER, 'rt')
+    >>> ft.read_timestamp()
+    <GSBRawdumpHeader pc: 2015 04 27 18 45 00 0.000000240>
```

Comment:
This is actually weird: you have 9 digits here, while above you state there are only 6.

Reply:
Ahh, this is a mistake, and Sanjay confirmed rawdump only writes GPS time.  I've made the documentation self-consistent now.